Source/MLXOptimizers/Optimizers.swift

```diff
@@ -654,3 +654,43 @@ open class Adafactor: OptimizerBase<Adafactor.State> {
         return (parameter - update, state)
     }
 }
+
+/// Clips the global norm of the gradients.
+///
+/// This function ensures that the global norm of the gradients does not exceed
+/// `max_norm`. It scales down the gradients proportionally if their norm is
+/// greater than `max_norm`.
+///
+/// - Parameters:
+///     - gradients: an array of MLXArray
+///     - maxNorm: the maximum allowed global norm of th gradients
+public func clipGradNorm(gradients: [MLXArray], maxNorm: Float) -> ([MLXArray], MLXArray) {
```

Comment:
I don't think we need this overload.. did you add it for a specific reason?

Reply:
Yes -- the python code is dealing with trees it looks like (and `ModuleParameters` is the swift equivalent) but a some of the swift code is dealing with flat arrays that haven't been recast into the tree form.

I don't have a caller yet, but I thought it might be convenient to support both.

If you think we don't need it I can easily remove it (and we can add it later if there is a call for it).